tuf/store/memorystore.go

```diff
@@ -54,6 +54,11 @@ func (m *memoryStore) SetMultiMeta(metas map[string][]byte) error {
 	return nil
 }
 
+func (m *memoryStore) DeleteMeta(name string) error {
```

Comment:
Maybe also add some  comments like `DeleteMeta()` on `HTTPStore` and `FilesystemStore`?